.github/container/gcp-tcpx-check.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+if [ -z "${DISABLE_TCPX_CHECK}" ]; then
+# Colors
+YELLOW='\033[0;33m'
+NOCOLOR='\033[0m'
+
+# Paths
+FILE="libnccl.so"
+DIR1="/var/lib/tcpx/lib64"
+DIR2="/usr/local/tcpx/lib64"
+
+# Attempt to retrieve the instance ID from the GCP metadata server
+INSTANCE_ID=$(curl  -m 1 -s -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/id" || true)
```

Comment:
Can we use the existence of the metadata server as the predicate here?

Reply:
What do you mean? Like just check if `google.internal` gets a dns hit? Probably a good idea actually...

Reply:
Done.